lib/dao.js

```diff
@@ -1169,6 +1170,14 @@ DataAccessObject._coerce = function (where) {
                 throw err;
               }
               break;
+            case 'regexp':
+              try {
+                val = utils.toRegExp(val);
+              } catch (err) {
+                err.statusCode = 400;
+                throw err;
```

Comment:
Let's try to avoid try/catch which will prevents V8 from optimizing the function. Can we return the Error from `utils.toRegExp(val);`?